src/airobot/arm/single_arm_pybullet.py

```diff
@@ -320,6 +321,52 @@ def move_ee_xyz(self, delta_xyz, eef_step=0.005, *args, **kwargs):
             success = self.set_jpos(jnt_poss, **kwargs)
         return success
 
+    def rot_ee_xyz(self, angle, axis='x', N=50, *args, **kwargs):
+        """
+        Rotate the end-effector about one of the end-effector axes,
+        without changing the position
+
+        Args:
+            angle (float): Angle by which to rotate in radians.
+            axis (str): Which end-effector frame axis to rotate about.
+                Must be in ['x', 'y', 'z'].
+            N (int): Number of waypoints along the rotation trajectory
+                (larger N means motion will be more smooth but potentially
+                slower).
+        Returns:
+            bool: A boolean variable representing if the action is successful
+            at the moment when the function exits.            
+        """
+        if not self._pb.in_realtime_mode():
+            raise AssertionError('rot_ee_xyz() can  '
+                                 'only be called in realtime'
+                                 ' simulation mode')
+
+        axis_dict = {'x': 0, 'y': 1, 'z': 2}
+        if axis not in ['x', 'y', 'z']:
```

Comment:
maybe better to use `if axis not in axis_dict` to remove duplicate code?

Reply:
fixed in 2259946